### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by tarkaiev**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        if (rectangle.color == null || rectangle.width == null || rectangle.length == null
+                || color == null || width == null || length == null) {
+            return this.hashCode() == rectangle.hashCode();
+        }
+        return width.equals(rectangle.width) && length.equals(rectangle.length)
+                && color.equals(rectangle.color);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: see checklist: While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;).

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
...
+    }
+
+    @Override
+    public int hashCode() {
+        int primecode = 31;
+        int result = 1;
+        result = primecode * result + (width == null ? 0 : width);
+        result = primecode * result + (length == null ? 0 : length);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: see checklist: you need to use hashcodes of fields

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (tarkaiev) [reply src/main/java/core/basesyntax/Rectangle.java]: thought that need hashcodes only in non-numeric fields
fixed

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (tarkaiev) [reply src/main/java/core/basesyntax/Rectangle.java]: fixed

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**